src/main/scala/com/databricks/labs/overwatch/pipeline/Snapshot.scala

```diff
@@ -0,0 +1,365 @@
+    val finalSnapshotRootPath  = if (Array("migration","restore").contains(processType.toLowerCase())){
+      s"${snapshotRootPath}/global_share"
+    }else{
+      s"${snapshotRootPath}/data"
+    }
+
+    sourcesToSnap.map(dataset => {
+      val sourceName = dataset.name.toLowerCase
+      val sourcePath = dataset.tableLocation
+      val mode = dataset._mode
+      val immutableColumns = (dataset.keys ++ dataset.incrementalColumns).distinct
+      val targetPath = s"$finalSnapshotRootPath/$sourceName"
+      CloneDetail(sourcePath, targetPath, None, cloneLevel,immutableColumns,mode)
+    }).toSeq
+  }
+
+  private[overwatch] def tableToExclude (sourceToSnap : Array[PipelineTable], excludes: Option[String] = Some("")) : Array[PipelineTable] = {
+
+    val exclude = excludes match {
+      case Some(s) if s.nonEmpty => s
+      case _ => ""
+    }
+    val excludeList = exclude.split(",").map(_.trim)
+
+    val cleanExcludes = excludeList.map(_.toLowerCase).map(exclude => {
+      if (exclude.contains(".")) exclude.split("\\.").takeRight(1).head else exclude
+    })
+
+    sourceToSnap
+      .filter(_.exists()) // source path must exist
+      .filterNot(t => cleanExcludes.contains(t.name.toLowerCase))
+  }
+
+
+  private[overwatch] def incrementalSnap(
+                                          pipelineTables : Array[PipelineTable],
+                                          excludes: Option[String] = Some("")
+                                        ): Unit = {
+
+
+    val sourceToSnapFiltered = tableToExclude(pipelineTables,excludes)
+
+    val cloneSpecs = buildCloneSpecs("Deep",sourceToSnapFiltered)
+    snapStream(cloneSpecs)
+  }
+
+  private[overwatch] def snap(
+                               pipelineTables : Array[PipelineTable],
+                               cloneLevel: String = "DEEP",
+                               excludes: Option[String] = Some("")
+                             ): Unit= {
+    val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+    require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
+      s"$cloneLevel. CloneLevels supported are ${acceptableCloneLevels.mkString(",")}.")
+
+    val sourceToSnapFiltered = tableToExclude(pipelineTables,excludes)
+
+    val cloneSpecs = buildCloneSpecs(cloneLevel,sourceToSnapFiltered)
+    val cloneReport = Helpers.parClone(cloneSpecs)
+    val cloneReportPath = s"${snapshotRootPath}/clone_report/"
+    cloneReport.toDS.write.format("delta").mode("append").save(cloneReportPath)
+    logger.log(Level.INFO, s"Clone report has been generated to this path: ${cloneReportPath}")
+  }
+
+}
+
+object Snapshot extends SparkSessionWrapper {
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+  def validate(sourceETLDB : String,
+               targetPrefix : String,
+               snapshotType : String,
+               pipeline : String = "Bronze,Silver,Gold",
+               tablesToExclude : String = " ",
```

Comment:
Are you doing any validation on this param?

Reply:
For this param there is no validation as such. as this can be "" or any table name. and if we get any table name which is not part of database it will anyway skip it.